misc/docker-ci/check.mk

```diff
@@ -1,7 +1,8 @@
 CONTAINER_NAME=h2oserver/h2o-ci:ubuntu1604
 SRC_DIR=/h2o
 CHECK_MK=$(SRC_DIR)/misc/docker-ci/check.mk
-CMAKE_ARGS=
+# promoting warnings to errors help us detect issues without dealing with false positives generated by every compiler that people might use
+CMAKE_ARGS=-Werror=format CMAKE_CXX_FLAGS=-Werror=format
```

Comment:
Shouldn't this be `CMAKE_ARGS=CMAKE_C_FLAGS=-Werror=format ...`?

But moreover ...